client/src/Attendance/Attendance.js

```diff
@@ -0,0 +1,281 @@
+    setColumnDates(updatedDates)
+  }
+
+  const generateColumns = () => {
+    let cols = []
+    for (let i = 0; i < 7; i++) {
+      cols.push({
+        dataIndex: 'date' + i,
+        key: 'date' + i,
+        width: 398,
+        // eslint-disable-next-line react/display-name
+        title: () => (
+          <DatePicker
+            disabledDate={c => c && c.valueOf() > Date.now()}
+            onChange={(_, ds) => handleDateChange(i, ds)}
+            bordered={false}
+            placeholder={t('selectDate')}
+            style={{ width: '8rem ', color: '#004A6E' }}
+          />
+        ),
+        // eslint-disable-next-line react/display-name
+        render: (_, record) => {
+          return (
+            <AttendanceDataCell
+              record={record}
+              columnIndex={i}
+              updateAttendanceData={updateAttendanceData}
+            />
+          )
+        }
+      })
+    }
+
+    return [
+      {
+        title: t('childNameCap'),
+        dataIndex: 'name',
+        width: 250,
+        key: 'name',
+        // eslint-disable-next-line react/display-name
+        render: (_, record) => {
+          return (
+            <div>
+              <p className="text-lg mb-1">
+                {record.childName || record.child.childName}
+              </p>
+              <p className="flex flex-wrap mt-0.5">
+                {record.business || record.child.business}{' '}
+                <img className="mx-1" alt="ellipse" src={ellipse} />{' '}
+                {record.cNumber || record.child.cNumber}
+              </p>
+            </div>
+          )
+        }
+      },
+      ...cols
+    ]
+  }
+
+  const [columns, setColumns] = useState(generateColumns())
+
+  i18n.on('languageChanged', () => setColumns(generateColumns()))
+
+  const handleSave = async () => {
+    const attendanceBatch = Object.entries(attendanceData).flatMap(cv =>
+      cv[1]
+        .filter(v => Object.keys(v).length > 0)
+        .map((v, k) => {
+          if (Object.keys(v).includes('absence')) {
+            return { ...v, check_in: columnDates[k], child_id: cv[0] }
+          }
+
+          const timeRegex = /(1[0-2]|0?[1-9]):([0-5][0-9]) (am|pm)/
+          const parsedTime1 = v.check_in.match(timeRegex)
+          const parsedTime2 = v.check_out.match(timeRegex)
+          const currentDate = dayjs(columnDates[k])
+          let checkoutDate
+
+          if (
+            (parsedTime1[3] === 'am' &&
+              parsedTime2[3] === 'am' &&
+              Number(parsedTime1[1]) > Number(parsedTime2[1])) ||
+            (parsedTime1[3] === 'pm' && parsedTime2[3] === 'am')
+          ) {
+            checkoutDate = currentDate.add(1, 'day').format('YYYY-MM-DD')
+          } else {
+            checkoutDate = currentDate.format('YYYY-MM-DD')
+          }
+
+          return {
+            check_in: `${columnDates[k]} ${v.check_in}`,
+            check_out: `${checkoutDate} ${v.check_out}`,
+            child_id: cv[0]
```

Comment:
Should this also include the absence reason?

Reply:
I know this might be waiting on my absences ticket, so if so, you can totally disregard this comment.

Reply:
Oop, definitely is waiting on my absences ticket, will resolve.